beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/BlockOperationSelectorFactory.java

```diff
@@ -128,24 +135,54 @@ public BlockOperationSelectorFactory(
                   contributionPool.createSyncAggregateForBlock(
                       blockSlotState.getSlot(), parentRoot))
           .executionPayload(
-              () ->
-                  forkChoiceNotifier
-                      .getPayloadId(parentRoot, blockSlotState.getSlot())
-                      .thenApply(
-                          maybePayloadId -> {
-                            if (maybePayloadId.isEmpty()) {
-                              // TTD not reached
-                              return SchemaDefinitionsBellatrix.required(
-                                      spec.atSlot(blockSlotState.getSlot()).getSchemaDefinitions())
-                                  .getExecutionPayloadSchema()
-                                  .getDefault();
-                            } else {
-                              return executionEngineChannel
-                                  .getPayload(maybePayloadId.get(), blockSlotState.getSlot())
-                                  .join();
-                            }
-                          })
-                      .join());
+              payloadProvider(
+                  parentRoot,
+                  blockSlotState,
+                  () ->
+                      SchemaDefinitionsBellatrix.required(
+                              spec.atSlot(blockSlotState.getSlot()).getSchemaDefinitions())
+                          .getExecutionPayloadSchema()
+                          .getDefault(),
+                  (payloadId) ->
+                      executionEngineChannel
+                          .getPayload(payloadId, blockSlotState.getSlot())
+                          .join()));
+
+      if (isMevBoostEnabled || blinded) {
+        bodyBuilder.executionPayloadHeader(
+            payloadProvider(
+                parentRoot,
+                blockSlotState,
+                () ->
+                    SchemaDefinitionsBellatrix.required(
+                            spec.atSlot(blockSlotState.getSlot()).getSchemaDefinitions())
+                        .getExecutionPayloadHeaderSchema()
+                        .getDefault(),
+                (payloadId) ->
+                    executionEngineChannel
+                        .getPayloadHeader(payloadId, blockSlotState.getSlot())
+                        .join()));
```

Comment:
Can we log a ticket to follow up and get rid of this `.join()` please? It's not a small amount of work but it would be a lot safer and more inline with our usual threading model to be able to return the `SafeFuture` here (which then means following that up the call stack quite a bit).

Reply:
tracked here: #5259